### ISSUE TITLE
Don't rely on the `__call__` for writing the dvc.yaml

### ISSUE THREAD

**Issue by PythonFZ** [zincware/ZnTrack#135]
In the current version the `__call__` is used to write the DVC file and the `__init__(load=True)` used the load argument.
This makes sense, but it might be beneficial to include a method `Node.zntrack.load()` and `Node.zntrack.write_dvc` (do not use save) that can replace the `load=True` and `__call__` as exclusive methods. This could e.g. be the methods that are added by the `@Node` decorator to the respective methods, allowing for an argument in the `@Node(use_call=False, use_init=False)` to restrict to the actual methods. It should not replace them, because `Node(load=True)` is very useful!

**Comment by PythonFZ** [comment]
We could also have:

```py
@Node()
class HelloWorld:
    def __init__(self, param):
        self.param = param
        
HelloWorld(param="Lorem Ipsum")
HelloWorld.zntrack.write_dvc()
# and
HelloWorld(load=True)
```

using something like

```py
class Node:
    def __init__(self):
        self.cls = None
    def __call__(self, *args, **kwargs):
        if self.cls is None:
            self.cls = args[0]
        self.apply_decorator()
        return self.cls       
    
    def apply_decorator(self):
        setattr(self.cls, "__init__", self.init_deco(getattr(self.cls, "__init__")))
    
    def init_deco(self, func):
        def wrapper(cls, *args, **kwargs):
            try:
                if kwargs['load']:
                    print("Returning loaded Node")
                    # does not work like this, needs some creative thinking
                    return func(cls, *args, **kwargs)  
            except KeyError:
                print("Init Wrap")
                return func(cls, *args, **kwargs)
        return wrapper
```

**Comment by PythonFZ** [comment]
One could also add a classmethod that yields the loaded cls, like 

hello_world = HelloWorld.load(name=...) which could be handled by a parent class for autocompletion.

**Comment by PythonFZ** [comment]
will be fixed in #176
